### PR TITLE
[HOTFIX] Fix Broken Activity IDs

### PR DESCRIPTION
**Pull request by kroffo**
* **Tickets addressed:** N/A
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
Fix some broken activity instance IDs which were left as `long` instead of being converted to `ActivityInstanceId`.

## Verification
Tested locally, requesting test from UI developer who found the issue as well.

## Documentation
None.

## Future work
Hopefully none.

### CODE REVIEW

**Review by pcrosemurgy — APPROVED** [review]

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetActivityAction.java [hunk 1]
```diff
@@ -41,17 +42,17 @@ public GetActivityAction(final Connection connection) throws SQLException {
     this.statement = connection.prepareStatement(sql);
   }
 
-  public ActivityInstance get(final long planId, final long activityId)
+  public ActivityInstance get(final long planId, final ActivityInstanceId activityId)
```
Comment (mattdailis) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetActivityAction.java:45]: IntelliJ is telling me `GetActivityAction` is unused - is there a reason to keep it around, or can we delete it?

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/SimulatedActivityRecord.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
     Map<String, SerializedValue> parameters,
     Instant start,
     Duration duration,
-    Long parentId,
-    List<Long> childIds,
+    ActivityInstanceId parentId,
```
Comment (mattdailis) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/SimulatedActivityRecord.java:17]: Hmm... I thought the reason for SimulatedActivityRecord to exist was that it used postgres ids, which were distinct from merlin ids....

Does it still make sense for it to exist as a separate concept from SimulatedActivity?

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulatedActivitiesAction.java [hunk 3]
```diff
@@ -41,22 +41,22 @@ public GetSimulatedActivitiesAction(final Connection connection) throws SQLExcep
...
-      final var parentId = resultSet.getLong(3);
+      final var parentId = new ActivityInstanceId(resultSet.getLong(3));
       final var startOffset = parseOffset(resultSet, 4, simulationStart);
       final var start = simulationStart.toInstant().plus(startOffset.in(MICROSECONDS), ChronoUnit.MICROS);
       final var duration = parseOffset(resultSet, 5, start);
       final var attributes = parseActivityAttributes(resultSet.getCharacterStream(6));
       final var directiveId = attributes.getLeft();
       final var arguments = attributes.getRight();
-      final var initialChildIds = new ArrayList<Long>();
+      final var initialChildIds = new ArrayList<ActivityInstanceId>();
```
Comment (mattdailis) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulatedActivitiesAction.java:59]: I'm trying to understand how this fixes Chris's bug and drawing blanks...

**Review by mattdailis — COMMENTED** [review]
Just some comments expressing my general confusion. If you have ready answers, feel free to address them, otherwise give me some time to stew 😉 

EDIT: I guess we are kind of in a hurry since it's a HOTFIX, so if Chris approves, feel free to merge even if I'm still stewing.

**Comment by camargo** [comment]
Still encountering the same error in merlin server:
```
[qtp2050370411-29] WARN io.javalin.Javalin - Uncaught exception
java.lang.NullPointerException: Cannot invoke "gov.nasa.jpl.aerie.merlin.server.remotes.postgres.SimulatedActivityRecord.childIds()" because the return value of "java.util.HashMap.get(Object)" is null
	at gov.nasa.jpl.aerie.merlin.server.remotes.postgres.GetSimulatedActivitiesAction.lambda$get$0(GetSimulatedActivitiesAction.java:74)
	at java.base/java.util.HashMap.forEach(HashMap.java:1421)
	at gov.nasa.jpl.aerie.merlin.server.remotes.postgres.GetSimulatedActivitiesAction.get(GetSimulatedActivitiesAction.java:72)
	at gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PostgresResultsCellRepository.getSimulatedActivities(PostgresResultsCellRepository.java:280)
	at gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PostgresResultsCellRepository.getSimulationResults(PostgresResultsCellRepository.java:235)
	at gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PostgresResultsCellRepository.getSimulationState(PostgresResultsCellRepository.java:220)
	at gov.nasa.jpl.aerie.merlin.server.remotes.postgres.PostgresResultsCellRepository$PostgresResultsCell.get(PostgresResultsCellRepository.java:443)
	at gov.nasa.jpl.aerie.merlin.server.services.CachedSimulationService.getSimulationResults(CachedSimulationService.java:14)
	at gov.nasa.jpl.aerie.merlin.server.services.GetSimulationResultsAction.run(GetSimulationResultsAction.java:52)
	at gov.nasa.jpl.aerie.merlin.server.http.MerlinBindings.getSimulationResults(MerlinBindings.java:140)
```

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulatedActivitiesAction.java (hunk 3 above)
Comment (Twisol) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSimulatedActivitiesAction.java:59]: According to Chris, it doesn't 😇

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/SimulatedActivityRecord.java (hunk 2 above)
Comment (Twisol) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/SimulatedActivityRecord.java:17]: Without looking too deep, I lean toward agreeing with Matt on this -- the fact that it's a `long` is important to the Postgres binding layer, and obfuscating it doesn't help in this component. Any conversion to wrappers should probably happen wherever the `SimulatedActivityRecord` is converted into something else.

EDIT: I tried to make a terrible pun (e*long*ation of IDs), but the conversion goes the other way, so I'm just stupid :)

**Comment by kroffo** [comment]
Closing until the issue is resolved.

**Comment by kroffo** [comment]
This branch was a mess. What happened was the following:

I was working on a separate task, and wrongly thought I found some usages of `Long` that needed to be converted to `ActivityInstanceId` that were causing issues. At that same moment, Chris let me know of a bug he ran into in the same file I was looking at, and I immediately thought that it was caused by the issue I had seen, and I jumped on the "fix". Turns out I had actually implemented the code right the first time (last week), and while unrolling these changes I remembered making these decisions quite consciously. Mondays are hard. New PR coming.

**Pull request closed without merging**